app/javascript/crayons/MultiSelectAutocomplete/MultiSelectAutocomplete.jsx

```diff
@@ -0,0 +1,388 @@
+/* eslint-disable jsx-a11y/interactive-supports-focus, jsx-a11y/role-has-required-aria-props */
+// Disabled due to the linter being out of date for combobox role: https://github.com/jsx-eslint/eslint-plugin-jsx-a11y/issues/789
+import { h, Fragment } from 'preact';
+import { useEffect, useRef, useReducer } from 'preact/hooks';
+import { Icon, Button } from '@crayons';
+import { Close } from '@images/x.svg';
+
+const KEYS = {
+  UP: 'ArrowUp',
+  DOWN: 'ArrowDown',
+  ENTER: 'Enter',
+  ESCAPE: 'Escape',
+  DELETE: 'Backspace',
+  COMMA: ',',
+  SPACE: ' ',
+};
+
+const ALLOWED_CHARS_REGEX = /([a-zA-Z0-9])/;
+
+const reducer = (state, action) => {
+  switch (action.type) {
+    case 'setSelectedItems':
+      return {
+        ...state,
+        selectedItems: action.payload,
+        suggestions: [],
+        activeDescendentIndex: null,
+      };
+    case 'setSuggestions':
+      return {
+        ...state,
+        suggestions: action.payload,
+        activeDescendentIndex: null,
+      };
+    case 'updateEditState':
+      return {
+        ...state,
+        editValue: action.payload.editValue,
+        inputPosition: action.payload.inputPosition,
+      };
+    case 'setActiveDescendentIndex':
+      return { ...state, activeDescendentIndex: action.payload };
+    case 'setIgnoreBlur':
+      return { ...state, ignoreBlur: action.payload };
+  }
```

Comment:
Refactor (blocking): If no action type is matched, return the existing state.

```suggestion
    default:
      return state;      
  }  
```

Reply:
woops! done now 🙂